qemu/tests/eject_media.py

```diff
@@ -24,90 +25,78 @@ def run(test, params, env):
     :param env: Dictionary with test environment
     """
 
-    qemu_binary = utils_misc.get_qemu_binary(params)
-    if not utils_misc.qemu_has_option("qmp", qemu_binary):
-        logging.warn("qemu does not support qmp. Human monitor will be used.")
-    qmp_used = False
     vm = env.get_vm(params["main_vm"])
     vm.verify_alive()
 
     session = vm.wait_for_login(timeout=int(params.get("login_timeout", 360)))
     logging.info("Wait until device is ready")
     time.sleep(10)
-    if vm.monitor.protocol == "qmp":
-        qmp_used = True
+
+    @decorators.fail_on(virt_vm.VMDeviceError)
+    def eject_cdrom(device, force=False):
+        return vm.eject_cdrom(device, force=False)
+
+    @decorators.fail_on(virt_vm.VMDeviceError)
+    def change_media(device, target):
+        return vm.change_media(device, target)
+
+    def check_block(block):
+        return True if block in str(vm.monitor.info("block")) else False
```

Comment:
It is not a suitable way to check whether the given media is in the list of `info block`, because `str()` may not provide the right information when the data structure returned by `info()` has been changed.